validator/src/journal/block_store.rs

```diff
@@ -38,7 +38,21 @@ pub trait BlockStore: Sync + Send {
     fn iter<'a>(&'a self) -> Result<Box<Iterator<Item = Block> + 'a>, BlockStoreError>;
 }
 
-#[derive(Default)]
+/// A block store that has fast lookups for transactions and batches
+pub trait IndexedBlockStore: BlockStore {
```

Comment:
I think this makes more sense as a separate trait rather than an extension of the BlockStore trait. I would suggest:
```
pub trait BatchIndex {
  fn contains(&self, id: &str) -> bool;
  fn block_by_id(&self, id: &str) -> Option<Block>;
}
```
We will want this kind of indexing when extend batch and transaction indexing to the BlockManager so duplicates and dependencies can be checked correctly.